k6 browser: Display Metrics and new Options page (#1667)
grafana/k6-docs: docs/sources/next/using-k6-browser/options.md

---
title: 'Browser options'

> description: 'An overview of the different options to customize the behaviour when running browser tests.'

Reviewer: ```suggestion
description: 'An overview of the different options you can use to customize the browser module behavior when running browser tests.'
```

weight: 04
---

# Browser options

To enable browser testing, add the `browser` configuration within the `options` property of the [Scenario options](https://grafana.com/docs/k6/<K6_VERSION>/using-k6/scenarios/#options).

{{< code >}}

```shared-iterations
export const options = {
  scenarios: {
    foo: {
      executor: 'shared-iterations',
      options: {
        browser: {
          type: 'chromium',
        },
      },
    },
  },
};
```

```constant-vus
export const options = {
  scenarios: {
    foo: {
      executor: 'constant-vus',
      vus: 10,
      duration: '5m',
      options: {
        browser: {
          type: 'chromium',
        },
      },
    },
  },
};
```

{{< /code >}}

## Script options

| Option                    | Description                                                         |
| ------------------------- | ------------------------------------------------------------------- |
| type<sup>(required)</sup> | Name of the browser running the test. Options include `'chromium'`. |

## Environment variable options

You can customize the behavior of the browser module by passing environment variables.

> {{% admonition type="note" %}}

Reviewer: ```suggestion
{{< admonition type="note" >}}
```

Customizing browser options via environment variables is unsupported when [running browser tests in Grafana Cloud k6](https://grafana.com/docs/grafana-cloud/testing/k6/get-started/run-your-first-browser-tests/).

> {{% /admonition %}}

Reviewer: ```suggestion
{{< /admonition >}}
```

{{< docs/shared source="k6" lookup="browser/env-var-options.md" version="<K6_VERSION>" >}}

### Default arguments

List of default arguments included when launching the browser process. You can pass one or more of the arguments to the `K6_BROWSER_IGNORE_DEFAULT_ARGS` environment variable when starting a test for the ones you want to ignore.

> {{% admonition type="note" %}}

Reviewer: ```suggestion
{{< admonition type="note" >}}
```

The starting '--' have been omitted from the argument names in these lists.

> {{% /admonition %}}

Reviewer: ```suggestion
{{< /admonition >}}
```
